clientv3/client.go

```diff
@@ -105,6 +105,43 @@ func (c *Client) SetEndpoints(eps ...string) {
 	c.balancer.updateAddrs(eps)
 }
 
+// Sync synchronizes client's endpoint.
+func (c *Client) Sync() error {
+	ctx, cancel := context.WithTimeout(context.Background(), c.cfg.DialTimeout)
```

Comment:
i do not think we should use dial timeout. sync should take a context like other calls. Sync(ctx context.Context)